sphinx/environment/__init__.py

```diff
@@ -668,6 +669,10 @@ def read_doc(self, docname, app=None):
         self.settings['trim_footnote_reference_space'] = \
             self.config.trim_footnote_reference_space
         self.settings['gettext_compact'] = self.config.gettext_compact
+        language = (self.config.language or 'en').replace('_', '-')
+        self.settings['language_code'] = language
+        if language in smartchars.quotes:
+            self.settings['smart_quotes'] = self.config.html_use_smartypants
```

Comment:
Note:
Regarding our deprecation policy, we should show deprecation message during single major version (1.6 or 1.7) like following.
https://github.com/sphinx-doc/sphinx/blob/master/sphinx/builders/latex.py#L261